src/main/java/de/azapps/kafkabackup/sink/BackupSinkTask.java

```diff
@@ -53,7 +53,7 @@ public void start(Map<String, String> props) {
 
         switch (config.storageMode()) {
             case S3:
-                offsetSink = new S3OffsetSink(config.bucketName(), adminClient);
+                offsetSink = new S3OffsetSink(config.bucketName(), adminClient, awsS3Service);
```

Comment:
Should we perhaps move this line to after initializing the awsS3Service? Even if it somehow magically works out by pointers / references, it seems scary to pass in null and expect the S3OffsetSink to not use it until its initialized?